monitoring/finished_build_stats_publisher.py

```diff
@@ -0,0 +1,18 @@
+import time
+
+
+class FinishedBuildStatsPublisher(object):
+
+    def __init__(self, collector, publisher):
+        self.collector = collector
+        self.publisher = publisher
+
+    def collect_and_publish(self, delay_ms=1000):
+        while True:
+            stats = self.collector.collect()
+            if stats.status != 'IN_PROGRESS':
```

Comment:
We think that instead of this condition, stats object should have a method `is_finished()`, which should be used here.